packages/typespec-autorest-canonical/README.md

```diff
@@ -70,3 +76,29 @@ Omit unreachable types. By default all types declared under the service namespac
 
 If the generated openapi types should have the `x-typespec-name` extension set with the name of the TypeSpec type that created it.
 This extension is meant for debugging and should not be depended on.
+
+## Decorators
+
+### Autorest
+
+- [`@useRef`](#@useref)
```

Comment:
I don't think this is a good way of dealing this. This will require adding this decorator to every common element in the library which we don't want to do. We are also going to be removing that from arm as we cannot have arm depend on autorest.

Instead you should build a set of known types and when resolving the type replace the reference

Reply:
removed the decorator